src/plugins/pddl-planner/pddl-planner_thread.cpp

```diff
@@ -235,13 +279,91 @@ PddlPlannerThread::dbmp_planner()
 	}
 }
 
+/**
+ * @brief Use Kstar planner for generating the top k plans for the given pddl planning problem
+ * 
+ */
+void
+PddlPlannerThread::kstar_planner()
+{
+	logger->log_info(name(), "Starting PDDL Planning with Fast-Downward: KStar...");
+
+	std::string command =
+	  "fast-downward.py" + std::string(" ") + cfg_domain_path_ + std::string(" ") + cfg_problem_path_;
+
+	if (!cfg_fd_options_.empty()) {
+		command += std::string(" ") + cfg_kstar_options_;
+	}
+	fs::remove_all("found_plans");
+
+	std::string result = run_planner(command);
+
+	size_t end_pos = result.find("Number of plans found: ");
+
+	if (end_pos == std::string::npos) {
+		logger->log_error(
+		  name(),
+		  "Expected \"Number of plans found: \" at the end of planner output but did not found");
+		throw Exception("Unexpected planner output");
+	}
+	int    plan_count = 0;
+	size_t new_line   = result.find("\n", end_pos);
+	plan_count        = std::stoi(result.substr(end_pos + 22, new_line - (end_pos + 22)));
```

Comment:
Where is that `22` coming from? Is this `std::string{"Number of plans found: "}.length()`? If so:
```suggestion
        const size_t count_offset = std::string{"Number of plans found: "}.length();
       	plan_count        = std::stoi(result.substr(end_pos + count_offset, new_line - (end_pos + cpunt_offset)));
```
Or possibly even:
```
constexpr count_offset = std::char_traits<char>::length("Number of plans found: ");
```
(see https://en.cppreference.com/w/cpp/string/char_traits/length)